tests/mmm/test_delayed_saturated_mmm.py

```diff
@@ -254,6 +258,49 @@ def test_get_fourier_models_data(
             assert fourier_modes_data.max().max() <= 1
             assert fourier_modes_data.min().min() >= -1
 
+    def test_get_channel_contributions_forward_pass_grid(
+        self, mmm_fitted: DelayedSaturatedMMM
+    ) -> None:
+        n_channels = len(mmm_fitted.channel_columns)
+        data_range = mmm_fitted.X.shape[0]
+        draws = 3
+        chains = 2
+        grid_size = 2
+        contributions = mmm_fitted.get_channel_contributions_forward_pass_grid(
+            start=0, stop=1.5, num=grid_size
+        )
+        assert contributions.shape == (
+            grid_size,
+            chains,
+            draws,
+            data_range,
+            n_channels,
+        )
```

Comment:
Could we test some simple numerical cases?

Reply:
Because this is an extremely complicated model, I would suggest providing a fake_fit, and pre-computed results from the forward pass. Just make sure those seem correct! This will be a regression test and not for the correctness of the output, but that's already something.

The reason for suggesting the extra-work is that this is the sort of place where it's easy for bugs to creep in without us realizing for a long time.

Reply:
Yes! You are right! I will improve this test! Thanks for the suggestions!

Reply:
@ricardoV94 I added more detailed tests that test the expected output of the forward pass. Let me know what you think about it!